### ISSUE TITLE
Version of 2018-07-14 does not compile on BASIS 750SP11

### ISSUE THREAD

**Issue by Grossjo1** [larshp/abapGit#1643]
Hi,

Verison 2017-12-13 is fine. 

Now i upload version 2018-07-14 into report ZJG_ABAPGIT.

On activation i get a short dump:
Category               Internal kernel error
Runtime Errors         GEN_ILLEGAL_SWITCH
Application Component  Not assigned
Date and Time          16.07.2018 10:25:51
[rabax.txt](https://github.com/larshp/abapGit/files/2197097/rabax.txt)

**Comment by flaiker** [comment]
Just tried the latest version on 740SP6, which worked fine. Can you try again? The dump also doesn't really tell anything unfortunately.

**Comment by christianguenter2** [comment]
Cannot reproduce this with our 750 SP11 system. But we have kernel 749, you have 745.
![image](https://user-images.githubusercontent.com/17437789/42753282-b32b9096-88f0-11e8-8b5a-4f0389c952af.png)

**Comment by larshp** [comment]
Tested ok in 750sp10, kernel release 749

The code in the dump looks strange, it is not part of https://raw.githubusercontent.com/abapGit/build/master/zabapgit.abap

```
|53045|CLASS ltcl_path DEFINITION FOR TESTING RISK LEVEL HARMLESS DURATION SHORT FINAL             |
|53046|  INHERITING FROM cl_aunit_assert.                                                          |
|53047|                                                                                            |
|53048|  PUBLIC SECTION.                                                                           |
|53049|    METHODS is_root FOR TESTING.                                                            |
|53050|    METHODS split_file_location FOR TESTING.                                                |
|53051|    METHODS is_subdir FOR TESTING.                                                          |
|53052|    METHODS change_dir FOR TESTING.                                                         |
|53053|    METHODS get_filename_from_syspath FOR TESTING.                                          |
|53054|                                                                                            |
|53055|ENDCLASS.   "ltcl_path    
```

**Comment by Grossjo1** [comment]
Hi,

tried it a few seconds ago. Shortdump still occurs
 but now the Sorce Code Extract refers to method zcl_abapgit_object_cmpt->zif_abapgit_object~changed_by( ).

on call 
CALL METHOD lo_cmp_template->('IF_CMP_TEMPLATE_EDIT~GET_CHANGE_USER')
          RECEIVING
            r_user = rv_user.
[rabax1.txt](https://github.com/larshp/abapGit/files/2197485/rabax1.txt)

**Comment by christianguenter2** [comment]
Hm, in both dumps occur on line 53073. I think that's no coincidence. Maybe you should raise an issue at SAP?

**Comment by Grossjo1** [comment]
HI even after some code rework (creating includes) the shortdump exists on the same line now with another codeline 47thousand something. 
I created an incident to the ABAP-Language-group, they are working on it already.
I will keep you informed.

**Comment by Grossjo1** [comment]
Hi Christian,

I did a code change with a technique which often helped from the old days back in the 90s using R2 4.3, 
Take out self containing code into an include include and activate them separately. Sometimes this helped in cases of activation.

But in the mean time BC-Developers and me found out the real problem:

in above mentioned code this snippet is called:

  CALL METHOD ('CL_CMP_TEMPLATE')=>('S_CREATE_FROM_DB') 
   EXPORTING
     **i_name = |{ ms_item-obj_name }|**
     ...

after changing the dynamic part to
        CALL METHOD ('CL_CMP_TEMPLATE')=>('S_CREATE_FROM_DB')
          EXPORTING
            **i_name         = ms_item-obj_name**
            i_version      = 'A'
          RECEIVING
            r_ref_template = lo_cmp_template.

the kernel 745 could activate the report. Kernel development is on its way for a correction.
Kind regards;
Joachim

**Comment by christianguenter2** [comment]
Coo.l Thank you for the info. The dynamic code is needed to be downward compatible. Because those APIs aren't available in lower releases.

**Comment by larshp** [comment]
suggest we incorporate the change described above, assuming it works with regards to typing. I guess the string template was added to convert the value

**Comment by larshp** [comment]
@Grossjo1 try again, latest version, @christianguenter2 has done the changes according to your findings

**Comment by Grossjo1** [comment]
Hi Christian,

This version works like a charm;-)

One i got the kernel patch wchich is in creation right now i will inform you.

Kind regards;
Joachim

**Comment by Grossjo1** [comment]
HI,
just got the info that the **not yet released** note 2673327 will cover that issue.

Kind regards;
